airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -1476,6 +1476,17 @@ def _merge_stream_slicers(
                     stream_cursor=cursor_component,
                 )
         elif model.incremental_sync:
+            if model.retriever.type == "AsyncRetriever":
+                if model.incremental_sync.type != "DatetimeBasedCursor":
+                    # TODO explain why it isn't supported
```

Comment:
TODO: I'm not exactly sure why other types of cursors wouldn't be supported but I was only doing this for source-amazon-ads so I wanted to be more restrictive than not.

Note that Global/PerPartition cursors were not updated which we will need for source-amazon-ads